main/integration-tests/support/complex/delete-study.js

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable no-console */
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const { processInBatches } = require('@aws-ee/base-services/lib/helpers/utils');
+
+const { parseS3Arn } = require('../aws/utils/s3-arn');
+const { run } = require('../utils/utils');
+
+/**
+ * A function that performs the complex task of deleting a study.
+ *
+ * @param {string} id The study id to be deleted
+ */
+async function deleteStudy({ aws, id = '' }) {
+  // The clean up logic is as follows:
+  // - As a measurement of caution, we only delete the study if its id contains the runId, this help us
+  //   avoid deleting anything by accident.
+  // - We get the study permission entry for the study from the study permissions table
+  // - We find all the users who have any access to the study
+  // - For each user, we find the user entry in the study permissions table
+  //   - We remove any entry for the study for the user
+  // - Delete the study entry from the study permissions table
+  // - We delete the study from the studies table
+  // - We delete the study s3 folder in the default data bucket (if the path contains runId)
+  //   We need to revisit this logic once the BYOB branch is merged
+
+  const runId = aws.settings.get('runId');
+
+  if (!id.includes(runId)) {
+    console.log(
+      `Study id "${id}" does not contain the runId "${runId}", skipping the deletion of this study as a measurement of caution`,
```

Comment:
Could we look for `-${id}-` instead, this would avoid unlikely deletion mistakes for a study named like 'StudyFor`runId`Research'. (Also, currently we have all studies creating by the test having the string -test- in them as well. Maybe we can leverage that too here?)

Reply:
Could we have the studyIds populated in an array, and then have them passed to this cleanup logic instead? That way we're not trying to match test-created studyIds with those already in the DB.